services/memory_storage/memory_storage.cpp

```diff
@@ -9,12 +9,16 @@
 #include <services/collection/collection.hpp>
 #include "route.hpp"
 
-using namespace components::result;
+using namespace components::cursor;
 
 namespace services {
 
     using namespace services::memory_storage;
 
+    std::pmr::memory_resource* default_resource(){
```

Comment:
Not bad but I recommend to place it in core directory and use it everywhere instead of `std::pmr::get_default_resource()`